src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,182 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int VALUE_NOT_FOUND = -1;
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+        size = 0;
+        head = null;
+        tail = null;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (size == 0) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == 0) {
+            addFirst(value);
+        } else if (index == size) {
+            add(value);
+        } else {
+            Node<T> currentAtIndex = getObjByIndex(index);
+            insert(value, currentAtIndex);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return getObjByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> current = getObjByIndex(index);
+        T oldValue = current.value;
+        current.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> current = getObjByIndex(index);
+        size--;
+        return unlink(current);
     }
 
     @Override
     public boolean remove(T object) {
+        int index = getIndex(object);
+        if (isElementIndex(index)) {
+            remove(index);
+            return true;
+        }
         return false;
     }
```

Comment:
if we rename `getIndex` to `getNode` and it will return `Node<T>`, then we can just unlink returned node
because now we go through our list in `getIndex` method and after that in `getObjByIndex` (inside `remove` method)